plugin/db/util/driverutil.go

```diff
@@ -216,36 +217,35 @@ func beginMigration(ctx context.Context, executor MigrationExecutor, m *db.Migra
 		}
 	}
 
-	// Check if there is any higher version already been applied
-	if version, err := executor.CheckOutOfOrderVersion(ctx, tx, m.Namespace, m.Source, m.Version); err != nil {
+	latestSequence, err := executor.FindLatestSequence(ctx, tx, m.Namespace)
+	if err != nil {
 		return -1, err
-	} else if version != nil && len(*version) > 0 {
-		// Clickhouse will always return non-nil version with empty string.
-		return -1, common.Errorf(common.MigrationOutOfOrder, fmt.Errorf("database %q has already applied version %s which is higher than %s", m.Database, *version, m.Version))
-	}
-
-	// If the migration engine is VCS and type is not baseline and is not branch, then we can only proceed if there is existing baseline
-	// This check is also wrapped in transaction to avoid edge case where two baselinings are running concurrently.
```

Comment:
Are we getting rid of the CheckOutOfOrderVersion method since this is the only places using it.


Also the existing code has the transactional property described in the comment "This check is also wrapped in transaction to avoid edge case where two baselinings are running concurrently."

If we use FindMigrationHistoryList, we are likely to lose that.

Reply:
Done.